### DISCUSSION TITLE
Transient Simulation with Peridynamics Module

### DISCUSSION THREAD

**Discussion by ppandit95** [idaholab/moose#19608] [Q&A General] [answered]
Dear MOOSE Community,

I wish to run a transient simulation using peridynamics module wherein density needs to be provided to create the mass matrix but I couldn't find any test for that and the simulations carried out are quasi static in nature with varying BCs with time.So in this context I wish to know whether it is possible as of now to run the transient simulations with Peridynamics Module . 

Many Thanks
Pushkar

**Comment by Traiwit** [comment]
Hi @ppandit95, I was having the same problem as well, I don't think MOOSE does have a true quasi-static (reached the steady-state and move on to the next step)

So I follow this 'Load step' method: https://mooseframework.inl.gov/source/executioners/Transient.html
then pre-define all the BCs, and use [Controls] to activate each BC at each particular time-step (arbitrary) 
```
... BCs block ...
     [./stope1]
       type = PorousFlowPiecewiseLinearSinkTC
     variable = porepressure
     boundary = 'stope1'
     pt_vals = '0 1e9' # x coordinates defining g
     multipliers = '0 1e9' # y coordinates defining g
     flux_function =  1e5 #1e5 # Variable C
     v = min
     use_mobility = true
     fluid_phase = 0
     save_in = flux
     use_relperm = true
     [../]

     [./stope2]
       type = PorousFlowPiecewiseLinearSinkTC
     variable = porepressure
     boundary = 'stope2'
     pt_vals = '0 1e9' # x coordinates defining g
     multipliers = '0 1e9' # y coordinates defining g
     flux_function =  1e5 #1e5 # Variable C
     v = min
     use_mobility = true
     fluid_phase = 0
     use_relperm = true
     save_in = flux
     [../]



...Controls block...
  [stope1_control]
    type = ConditionalFunctionEnableControl
    enable_objects = 'BCs::stope1'
    conditional_function = 'if(t>=2 ,1,0)'
    execute_on = 'initial timestep_begin'
  []

  [stope2_control]
    type = ConditionalFunctionEnableControl
    enable_objects = 'BCs::stope2'
    conditional_function = 'if(t>=3  ,1,0)'
    execute_on = 'initial timestep_begin'
  []

[]

```

@GiudGiud I still would like to have a multi-steps steady-state solver (true quasi-SS), not sure if you can suggest it to the MOOSE team?

**Reply by ppandit95** [reply]
ohh,thats great I have been using time varying BCs but this also seems a alternative way to solve the problem,thanks @Traiwit

**Reply by GiudGiud** [reply]
So you're talking about nested relaxation transient solves inside a global transient?

Or nested direct steady solves inside a global transient?

**Reply by GiudGiud** [reply]
Either way, I think our rework of the Executioner system (currently ongoing) will help us nest solves like this. 
We ll keep you posted

**Reply by Traiwit** [reply]
@GiudGiud more like the 2nd case, but yes awesome, thank you!

**Comment by hugary1995** [comment, marked as answer]
Adding the `InertialForce` kernel is equivalent to creating the mass matrix you are talking about. In the Executioner, you can specify
```python
[TimeIntegrator]
  type = NewmarkBeta
[]
```
to use the Newmark-beta time integrator. Generalized-alpha methods are also available if you search the regression tests.

**Reply by ppandit95** [reply]
Thanks @hugary1995 for elaborating on the implementation of mass matrix and the relevant executioner.I will look into it...
